stratum/hal/lib/tdi/dpdk/dpdk_port_config.cc

```diff
@@ -0,0 +1,201 @@
+  }
+}
+
+::util::Status DpdkPortConfig::SetParam(
+    ValueCase value_case,
+    const SingletonPort& singleton_port) {
+
+  const auto& config_params = singleton_port.config_params();
+
+  switch (value_case) {
+    case ValueCase::kPortType:
+      params_set |= GNMI_CONFIG_PORT_TYPE;
+      cfg.port_type = config_params.type();
+      LOG(INFO) << "PortParam::kPortType = " << cfg.port_type;
+      break;
+
+    case ValueCase::kDeviceType:
+      params_set |= GNMI_CONFIG_DEVICE_TYPE;
+      cfg.device_type = config_params.device_type();
+      LOG(INFO) << "PortParam::kDeviceType = " << cfg.device_type;
+      break;
+
+    case ValueCase::kQueueCount:
+      params_set |= GNMI_CONFIG_QUEUE_COUNT;
+      cfg.queues = config_params.queues();
+      LOG(INFO) << "PortParam::kQueueCount = " << cfg.queues;
+      break;
+
+    case ValueCase::kSockPath:
+      params_set |= GNMI_CONFIG_SOCKET_PATH;
+      cfg.socket_path = config_params.socket();
+      LOG(INFO) << "PortParam::kSockPath = " << cfg.socket_path;
+      break;
+
+    case ValueCase::kHostConfig:
+      params_set |= GNMI_CONFIG_HOST_NAME;
+      cfg.host_name = config_params.host_name();
+      LOG(INFO) << "PortParam::kHostConfig = " << cfg.host_name;
+      break;
+
+    case ValueCase::kPipelineName:
+      cfg.pipeline_name = config_params.pipeline();
+      params_set |= GNMI_CONFIG_PIPELINE_NAME;
+      LOG(INFO) << "PortParam::kPipelineName = " << cfg.pipeline_name;
+      break;
+
+    case ValueCase::kMempoolName:
+      cfg.mempool_name = config_params.mempool();
+      params_set |= GNMI_CONFIG_MEMPOOL_NAME;
+      LOG(INFO) << "PortParam::kMempoolName = " << cfg.mempool_name;
+      break;
+
+    case ValueCase::kControlPort:
+      control_port = config_params.control();
+      LOG(INFO) << "PortParam::kControlPort = " << control_port;
+      break;
+
+    case ValueCase::kPciBdf:
+      params_set |= GNMI_CONFIG_PCI_BDF_VALUE;
+      cfg.pci_bdf = config_params.pci();
+      LOG(INFO) << "PortParam::kPciBdf = " << cfg.pci_bdf;
+      break;
+
+    case ValueCase::kMtuValue:
+      if (config_params.mtu() > MAX_MTU) {
+        return MAKE_ERROR(ERR_INVALID_PARAM)
+             << "Unsupported MTU = " << config_params.mtu()
+             << ". MTU should be less than " << MAX_MTU << ".";
+      }
+      mtu = config_params.mtu();
+      params_set |= GNMI_CONFIG_MTU_VALUE;
+      LOG(INFO) << "PortParam::kMtuValue = " << mtu.value();
+      break;
+
+    case ValueCase::kPacketDir:
+      params_set |= GNMI_CONFIG_PACKET_DIR;
+      cfg.packet_dir = config_params.packet_dir();
+      LOG(INFO) << "PortParam::kPacketDir = " << cfg.packet_dir;
+      break;
+
+    default:
+      break;
+  }
+  return ::util::OkStatus();
+}
+
+::util::Status DpdkPortConfig::SetHotplugParam(
+  SWBackendHotplugParams param_type, const SingletonPort& singleton_port) {
+
+  const auto& params = singleton_port.config_params().hotplug_config();
+  auto& hotplug = cfg.hotplug_config;
```

Comment:
same comment regarding cfg

Reply:
As far as I can tell, there's no reason for `HotplugConfigParams` to be a member of `PortConfigParams`. The two sets of parameters appear to be orthogonal. I'm tempted to make them peers.